### ISSUE TITLE
Duplication of the displayed table

### ISSUE THREAD

**Issue by ZohaSpark** [metabase/metabase#17060]
**Describe the bug**
On some questions, the table is replicated right next to the original one which creates quite long rows. It does not come from the code because I tried replicating the question as a new one and it is displayed normally.

**To Reproduce**
I am not sure on how to reproduce this behavior.

**Expected behavior**
Have a table displaying each column only once.

**Screenshots**

**The wrong display, the settings available and the "raw" display confirming the double columns:**
![image](https://user-images.githubusercontent.com/87467508/125763003-65753237-ea07-4835-8505-0fff35caee42.png)
![image](https://user-images.githubusercontent.com/87467508/125763290-c5a1f23d-3c5a-4023-ad5f-c22cbc014f5f.png)
![image](https://user-images.githubusercontent.com/87467508/125763632-5278043a-ea4b-48b9-95d6-5c68cb89323f.png)

**The other version I created:**
![image](https://user-images.githubusercontent.com/87467508/125763457-50184c1e-1a74-4983-b69d-823850569996.png)
![image](https://user-images.githubusercontent.com/87467508/125763514-15fef8e2-2c40-4ae1-875b-d48a0ee94153.png)

**Information about your Metabase Installation:**

> {
  "browser-info": {
    "language": "fr-FR",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.11+9",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.11",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.11+9",
    "os.name": "Linux",
    "os.version": "4.14.232-177.418.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "redshift"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.9"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.18"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-07-08",
      "tag": "v0.40.0",
      "branch": "master",
      "hash": "961caed"
    },
    "settings": {
      "report-timezone": "Europe/Brussels"
    }
  }
}

**Severity**
It is quite annoying and time-consuming. We need to copy & paste it onto a new question and set again our filters and their specifications. It is also a frequent issue, making it worse for us. On the other hand, for now it is not destroying our abilities to work with Metabase.

**Comment by flamber** [comment]
Hi @ZohaSpark 
Is this a new problem? When did it start?
I would guess that you might be able to "reset" a bad question, by going into the Notebook editor, clicking the Columns-link on the right and Select All, but before doing that, can you go into one of the problematic questions, note the ID in the URL, and then find the row in the application database table `report_card` and post the entire row here?

**Comment by ZohaSpark** [comment]
Hi @flamber , so I noticed this on Tuesday, but I could not say when it exactly started. I will show you the 1664 question, that has that bug:

**id: 1664
created_at: July 13, 2021, 13:45
updated_at: July 15, 2021, 12:20
name: 1: performances by campaign product | comparison
description: empty
display: table**


**dataset_query:**

> {
  "native": {
    "template-tags": {
      "date2": {
        "id": "de3a3042-8a92-110c-9d34-872aeffae36d",
        "name": "date2",
        "display-name": "Date2",
        "type": "dimension",
        "dimension": [
          "field",
          46136,
          null
        ],
        "widget-type": "date/all-options",
        "required": true
      },
      "brand_included": {
        "id": "841a1446-bcb4-ce89-cebe-e6fca305020c",
        "name": "brand_included",
        "display-name": "brand included",
        "type": "dimension",
        "dimension": [
          "field",
          789,
          null
        ],
        "widget-type": "category"
      },
      "device": {
        "id": "d57f2d34-8a39-e4d8-572e-c78b15590ae5",
        "name": "device",
        "display-name": "device",
        "type": "dimension",
        "dimension": [
          "field",
          541,
          null
        ]
      },
      "account_name": {
        "id": "63f8f91e-de96-63a9-ff1a-db184b8e546f",
        "name": "account_name",
        "display-name": "Account name",
        "type": "dimension",
        "dimension": [
          "field",
          53944,
          null
        ],
        "widget-type": "category"
      },
      "conversion_type": {
        "id": "64ab1409-cfd0-cac1-93e5-134dbc69163e",
        "name": "conversion_type",
        "display-name": "conversion_type",
        "type": "dimension",
        "dimension": [
          "field",
          544,
          null
        ],
        "widget-type": "category"
      },
      "event": {
        "id": "fcf399ee-b20f-389d-545b-f8cc1473d1cf",
        "name": "event",
        "display-name": "event",
        "type": "dimension",
        "dimension": [
          "field",
          543,
          null
        ]
      },
      "legacy": {
        "id": "d40607d6-6b45-a714-a6ad-08eecad34874",
        "name": "legacy",
        "display-name": "legacy",
        "type": "dimension",
        "dimension": [
          "field",
          551,
          null
        ],
        "widget-type": "category"
      },
      "product": {
        "id": "0861c1f4-906f-a68d-e821-c2453d8e3c93",
        "name": "product",
        "display-name": "product",
        "type": "dimension",
        "dimension": [
          "field",
          550,
          null
        ],
        "widget-type": "category"
      },
      "clients": {
        "id": "3b748eaa-687e-8fc6-17bd-b9cb2ed47486",
        "name": "clients",
        "display-name": "clients",
        "type": "dimension",
        "dimension": [
          "field",
          54004,
          null
        ],
        "widget-type": "category",
        "required": true
      },
      "network": {
        "id": "641fb3d2-9a88-6d60-0083-1dd23310c832",
        "name": "network",
        "display-name": "network",
        "type": "dimension",
        "dimension": [
          "field",
          545,
          null
        ],
        "widget-type": "category"
      },
      "location": {
        "id": "734326d8-a971-7106-593a-a63675dd53c4",
        "name": "location",
        "display-name": "location",
        "type": "dimension",
        "dimension": [
          "field",
          554,
          null
        ]
      },
      "type": {
        "id": "0e622fea-4790-164b-057a-bd85b53d546b",
        "name": "type",
        "display-name": "type",
        "type": "dimension",
        "dimension": [
          "field",
          546,
          null
        ],
        "widget-type": "category"
      },
      "other": {
        "id": "26672a37-6fae-2e12-7da3-4185f5299013",
        "name": "other",
        "display-name": "other filter",
        "type": "dimension",
        "dimension": [
          "field",
          549,
          null
        ]
      },
      "date": {
        "id": "4e3b89a8-063f-f135-4ae7-5491eda0504a",
        "name": "date",
        "display-name": "date",
        "type": "dimension",
        "dimension": [
          "field",
          46136,
          null
        ],
        "widget-type": "date/all-options",
        "required": true
      },
      "channel": {
        "id": "54fff1d7-ff2d-21cd-2c58-7af0fa437796",
        "name": "channel",
        "display-name": "channel",
        "type": "dimension",
        "dimension": [
          "field",
          553,
          null
        ],
        "widget-type": "category"
      },
      "category": {
        "id": "ab3aa140-ca0d-2827-e4b6-00af936d04e6",
        "name": "category",
        "display-name": "category",
        "type": "dimension",
        "dimension": [
          "field",
          548,
          null
        ]
      },
      "language": {
        "id": "62373f98-d51d-b416-1fbe-65aaca495f7d",
        "name": "language",
        "display-name": "language",
        "type": "dimension",
        "dimension": [
          "field",
          547,
          null
        ],
        "widget-type": "category"
      }
    },
    "query": "SELECT * FROM \r\n(\r\n    SELECT \r\n        MIN(date) AS start_date,\r\n        MAX(date) AS end_date,\r\n        2 AS rank,\r\n        LOWER(reports.campaign_details.campaign_product) AS campaign_product,\r\n        SUM(clicks) AS clicks,\r\n        SUM(client_cost)/nullif(SUM(clicks),0)::decimal AS cpc,\r\n        SUM(client_cost) AS client_cost,\r\n        SUM(conversions) AS conversions,\r\n        SUM(conversions)/nullif(SUM(clicks),0)::decimal AS conversion_rate,\r\n        SUM(client_cost)/nullif(SUM(conversions),0)::decimal AS cpa,\r\n        SUM(conversion_value)/nullif(SUM(conversions),0)::decimal AS avg_revenue,\r\n        SUM(conversion_value) AS revenue,\r\n        SUM(conversion_value)/nullif(SUM(client_cost),0)::decimal AS roas\r\n    FROM utils.clients\r\n        JOIN utils.client_ad_platforms ON utils.client_ad_platforms.client_id = utils.clients.id\r\n        JOIN reports.campaign_details ON reports.campaign_details.account_id = utils.client_ad_platforms.account_id\r\n        JOIN global.campaign_daily_metrics ON global.campaign_daily_metrics.campaign_id = reports.campaign_details.campaign_id,\r\n        LATERAL (SELECT textual_boolean AS brand_included FROM metabase_filters WHERE {{brand_included}} LIMIT 1) AS b\r\n    WHERE TRUE\r\n        [[AND {{clients}}]]\r\n        [[AND {{date}}]]\r\n        [[AND {{channel}}]]\r\n        [[AND {{network}}]]\r\n        [[AND {{type}}]]\r\n        [[AND {{category}}]]\r\n        [[AND {{location}}]]\r\n        [[AND {{language}}]]\r\n        [[AND {{product}}]]\r\n        [[AND {{device}}]]\r\n        [[AND {{event}}]]\r\n        [[AND {{conversion_type}}]]\r\n        [[AND {{other}}]]\r\n        [[AND {{legacy}}]]\r\n        [[AND {{account_name}}]]\r\n        AND CASE WHEN b.brand_included = 'no' THEN LOWER(coalesce(reports.campaign_details.campaign_type,'')) NOT LIKE '%brand%' AND LOWER(coalesce(reports.campaign_details.campaign_category,'')) NOT LIKE '%brand%'  ELSE 1 = 1 END\r\n    GROUP BY 3,4\r\nUNION\r\n    SELECT \r\n        MIN(date) AS start_date,\r\n        MAX(date) AS end_date,\r\n        1 AS rank,\r\n        'Total' AS campaign_product,\r\n        SUM(clicks) AS clicks,\r\n        SUM(client_cost)/nullif(SUM(clicks),0)::decimal AS cpc,\r\n        SUM(client_cost) AS client_cost,\r\n        SUM(conversions) AS conversions,\r\n        SUM(conversions)/nullif(SUM(clicks),0)::decimal AS conversion_rate,\r\n        SUM(client_cost)/nullif(SUM(conversions),0)::decimal AS cpa,\r\n        SUM(conversion_value)/nullif(SUM(conversions),0)::decimal AS avg_revenue,\r\n        SUM(conversion_value) AS revenue,\r\n        SUM(conversion_value)/nullif(SUM(client_cost),0)::decimal AS roas\r\n    FROM utils.clients\r\n        JOIN utils.client_ad_platforms ON utils.client_ad_platforms.client_id = utils.clients.id\r\n        JOIN reports.campaign_details ON reports.campaign_details.account_id = utils.client_ad_platforms.account_id\r\n        JOIN global.campaign_daily_metrics ON global.campaign_daily_metrics.campaign_id = reports.campaign_details.campaign_id,\r\n        LATERAL (SELECT textual_boolean AS brand_included FROM metabase_filters WHERE {{brand_included}} LIMIT 1) AS b\r\n    \r\n    WHERE TRUE\r\n        [[AND {{clients}}]]\r\n        [[AND {{date}}]]\r\n        [[AND {{channel}}]]\r\n        [[AND {{network}}]]\r\n        [[AND {{type}}]]\r\n        [[AND {{category}}]]\r\n        [[AND {{location}}]]\r\n        [[AND {{language}}]]\r\n        [[AND {{product}}]]\r\n        [[AND {{device}}]]\r\n        [[AND {{event}}]]\r\n        [[AND {{conversion_type}}]]\r\n        [[AND {{other}}]]\r\n        [[AND {{legacy}}]]\r\n        [[AND {{account_name}}]]\r\n        AND CASE WHEN b.brand_included = 'no' THEN LOWER(coalesce(reports.campaign_details.campaign_type,'')) NOT LIKE '%brand%' AND LOWER(coalesce(reports.campaign_details.campaign_category,'')) NOT LIKE '%brand%'  ELSE 1 = 1 END\r\n\r\n    GROUP BY 3,4\r\n\r\nUNION\r\n\r\n    SELECT \r\n        MIN(date) AS start_date,\r\n        MAX(date) AS end_date,\r\n        2 AS rank,\r\n        LOWER(reports.campaign_details.campaign_product) AS campaign_product,\r\n        SUM(clicks) AS clicks,\r\n        SUM(client_cost)/nullif(SUM(clicks),0)::decimal AS cpc,\r\n        SUM(client_cost) AS client_cost,\r\n        SUM(conversions) AS conversions,\r\n        SUM(conversions)/nullif(SUM(clicks),0)::decimal AS conversion_rate,\r\n        SUM(client_cost)/nullif(SUM(conversions),0)::decimal AS cpa,\r\n        SUM(conversion_value)/nullif(SUM(conversions),0)::decimal AS avg_revenue,\r\n        SUM(conversion_value) AS revenue,\r\n        SUM(conversion_value)/nullif(SUM(client_cost),0)::decimal AS roas\r\n    FROM utils.clients\r\n        JOIN utils.client_ad_platforms ON utils.client_ad_platforms.client_id = utils.clients.id\r\n        JOIN reports.campaign_details ON reports.campaign_details.account_id = utils.client_ad_platforms.account_id\r\n        JOIN global.campaign_daily_metrics ON global.campaign_daily_metrics.campaign_id = reports.campaign_details.campaign_id,\r\n        LATERAL (SELECT textual_boolean AS brand_included FROM metabase_filters WHERE {{brand_included}} LIMIT 1) AS b\r\n    WHERE TRUE\r\n        [[AND {{clients}}]]\r\n        [[AND {{date2}}]]\r\n        [[AND {{channel}}]]\r\n        [[AND {{network}}]]\r\n        [[AND {{type}}]]\r\n        [[AND {{category}}]]\r\n        [[AND {{location}}]]\r\n        [[AND {{language}}]]\r\n        [[AND {{product}}]]\r\n        [[AND {{device}}]]\r\n        [[AND {{event}}]]\r\n        [[AND {{conversion_type}}]]\r\n        [[AND {{other}}]]\r\n        [[AND {{legacy}}]]\r\n        [[AND {{account_name}}]]\r\n        AND CASE WHEN b.brand_included = 'no' THEN LOWER(coalesce(reports.campaign_details.campaign_type,'')) NOT LIKE '%brand%' AND LOWER(coalesce(reports.campaign_details.campaign_category,'')) NOT LIKE '%brand%'  ELSE 1 = 1 END\r\n    GROUP BY 3,4\r\nUNION\r\n    SELECT \r\n        MIN(date) AS start_date,\r\n        MAX(date) AS end_date,\r\n        1 AS rank,\r\n        'Total' AS campaign_product,\r\n        SUM(clicks) AS clicks,\r\n        SUM(client_cost)/nullif(SUM(clicks),0)::decimal AS cpc,\r\n        SUM(client_cost) AS client_cost,\r\n        SUM(conversions) AS conversions,\r\n        SUM(conversions)/nullif(SUM(clicks),0)::decimal AS conversion_rate,\r\n        SUM(client_cost)/nullif(SUM(conversions),0)::decimal AS cpa,\r\n        SUM(conversion_value)/nullif(SUM(conversions),0)::decimal AS avg_revenue,\r\n        SUM(conversion_value) AS revenue,\r\n        SUM(conversion_value)/nullif(SUM(client_cost),0)::decimal AS roas\r\n    FROM utils.clients\r\n        JOIN utils.client_ad_platforms ON utils.client_ad_platforms.client_id = utils.clients.id\r\n        JOIN reports.campaign_details ON reports.campaign_details.account_id = utils.client_ad_platforms.account_id\r\n        JOIN global.campaign_daily_metrics ON global.campaign_daily_metrics.campaign_id = reports.campaign_details.campaign_id,\r\n        LATERAL (SELECT textual_boolean AS brand_included FROM metabase_filters WHERE {{brand_included}} LIMIT 1) AS b\r\n    \r\n    WHERE TRUE\r\n        [[AND {{clients}}]]\r\n        [[AND {{date2}}]]\r\n        [[AND {{channel}}]]\r\n        [[AND {{network}}]]\r\n        [[AND {{type}}]]\r\n        [[AND {{category}}]]\r\n        [[AND {{location}}]]\r\n        [[AND {{language}}]]\r\n        [[AND {{product}}]]\r\n        [[AND {{device}}]]\r\n        [[AND {{event}}]]\r\n        [[AND {{conversion_type}}]]\r\n        [[AND {{other}}]]\r\n        [[AND {{legacy}}]]\r\n        [[AND {{account_name}}]]\r\n        AND CASE WHEN b.brand_included = 'no' THEN LOWER(coalesce(reports.campaign_details.campaign_type,'')) NOT LIKE '%brand%' AND LOWER(coalesce(reports.campaign_details.campaign_category,'')) NOT LIKE '%brand%'  ELSE 1 = 1 END\r\n\r\n    GROUP BY 3,4\r\n) AS unionquery\r\nWHERE client_cost > 0\r\nORDER BY rank, campaign_product DESC, start_date DESC"
  },
  "type": "native",
  "database": 3
}

**visualization_settings:**

> {
  "table.pivot_column": "start_date",
  "table.cell_column": "rank",
  "table.column_formatting": [
    {
      "columns": [
        "campaign_category",
        "campaign_product"
      ],
      "type": "single",
      "operator": "=",
      "value": "Total",
      "color": "#509EE3",
      "highlight_row": true
    },
    {
      "max_value": 100,
      "color": "#509EE3",
      "columns": [
        "cpa"
      ],
      "value": "",
      "type": "range",
      "colors": [
        "#84BB4C",
        "#F9CF48",
        "#ED6E6E"
      ],
      "highlight_row": false,
      "min_value": 0,
      "min_type": null,
      "operator": "=",
      "max_type": "custom"
    }
  ],
  "card.title": "Synthèse des performance par produit",
  "table.columns": [
    {
      "name": "rank",
      "enabled": false
    },
    {
      "name": "start_date",
      "fieldRef": [
        "field",
        "start_date",
        {
          "base-type": "type/Date"
        }
      ],
      "enabled": true
    },
    {
      "name": "end_date",
      "fieldRef": [
        "field",
        "end_date",
        {
          "base-type": "type/Date"
        }
      ],
      "enabled": true
    },
    {
      "name": "campaign_product",
      "enabled": true
    },
    {
      "name": "clicks",
      "enabled": true
    },
    {
      "name": "cpc",
      "enabled": true
    },
    {
      "name": "client_cost",
      "enabled": true
    },
    {
      "name": "conversions",
      "enabled": true
    },
    {
      "name": "conversion_rate",
      "enabled": true
    },
    {
      "name": "cpa",
      "enabled": true
    },
    {
      "name": "avg_revenue",
      "enabled": true
    },
    {
      "name": "revenue",
      "enabled": true
    },
    {
      "name": "roas",
      "enabled": true
    },
    {
      "name": "rank",
      "fieldRef": [
        "field",
        "rank",
        {
          "base-type": "type/Integer"
        }
      ],
      "enabled": false
    },
    {
      "name": "campaign_product",
      "fieldRef": [
        "field",
        "campaign_product",
        {
          "base-type": "type/Text"
        }
      ],
      "enabled": false
    },
    {
      "name": "clicks",
      "fieldRef": [
        "field",
        "clicks",
        {
          "base-type": "type/Decimal"
        }
      ],
      "enabled": false
    },
    {
      "name": "cpc",
      "fieldRef": [
        "field",
        "cpc",
        {
          "base-type": "type/Float"
        }
      ],
      "enabled": false
    },
    {
      "name": "client_cost",
      "fieldRef": [
        "field",
        "client_cost",
        {
          "base-type": "type/Float"
        }
      ],
      "enabled": false
    },
    {
      "name": "conversions",
      "fieldRef": [
        "field",
        "conversions",
        {
          "base-type": "type/Float"
        }
      ],
      "enabled": false
    },
    {
      "name": "conversion_rate",
      "fieldRef": [
        "field",
        "conversion_rate",
        {
          "base-type": "type/Float"
        }
      ],
      "enabled": false
    },
    {
      "name": "cpa",
      "fieldRef": [
        "field",
        "cpa",
        {
          "base-type": "type/Float"
        }
      ],
      "enabled": false
    },
    {
      "name": "avg_revenue",
      "fieldRef": [
        "field",
        "avg_revenue",
        {
          "base-type": "type/Float"
        }
      ],
      "enabled": false
    },
    {
      "name": "revenue",
      "fieldRef": [
        "field",
        "revenue",
        {
          "base-type": "type/Float"
        }
      ],
      "enabled": false
    },
    {
      "name": "roas",
      "fieldRef": [
        "field",
        "roas",
        {
          "base-type": "type/Float"
        }
      ],
      "enabled": false
    }
  ],
  "column_settings": {
    "[\"name\",\"conversion_3_proportion\"]": {
      "number_style": "percent",
      "column_title": "PNO"
    },
    "[\"name\",\"cac\"]": {
      "scale": 1,
      "decimals": 1,
      "number_style": "currency",
      "currency": "EUR",
      "currency_in_header": false
    },
    "[\"name\",\"total_score\"]": {
      "column_title": "Revenue",
      "decimals": 0,
      "number_style": "currency",
      "currency": "EUR",
      "currency_in_header": false
    },
    "[\"name\",\"conversions\"]": {
      "decimals": 0
    },
    "[\"name\",\"avg_conversion_value\"]": {
      "column_title": "Avg revenue",
      "scale": 1,
      "decimals": 0,
      "number_style": "currency",
      "currency": "EUR",
      "currency_in_header": false
    },
    "[\"name\",\"cpa\"]": {
      "column_title": "Cac",
      "decimals": 1,
      "number_style": "currency",
      "currency": "EUR",
      "currency_in_header": false
    },
    "[\"name\",\"cpc\"]": {
      "decimals": 1,
      "number_style": "currency",
      "currency": "EUR",
      "currency_in_header": false
    },
    "[\"name\",\"conversion_1_proportion\"]": {
      "column_title": "HU",
      "number_style": "percent"
    },
    "[\"name\",\"cost\"]": {
      "decimals": 0,
      "number_style": "currency",
      "currency": "EUR",
      "currency_in_header": false
    },
    "[\"name\",\"conversion_2_proportion\"]": {
      "column_title": "MRH",
      "number_style": "percent"
    },
    "[\"name\",\"conversion_rate\"]": {
      "number_style": "percent"
    },
    "[\"name\",\"start_date\"]": {
      "time_enabled": null
    },
    "[\"name\",\"end_date\"]": {
      "time_enabled": null
    }
  }
}

**creator_id: 76
databse_id:3
table_id: Empty
query_type: native
archived: false
collection_id: 6
public_uuid: Empty
made_public_by_id: Empty
enable_embedding: false
embedding_params: Empty
cache_ttl: Empty**


**result_metadata:**

> [
  {
    "name": "start_date",
    "display_name": "start_date",
    "base_type": "type/Date",
    "field_ref": [
      "field",
      "start_date",
      {
        "base-type": "type/Date"
      }
    ],
    "semantic_type": "type/CreationDate",
    "fingerprint": {
      "global": {
        "distinct-count": 2,
        "nil%": 0
      },
      "type": {
        "type/DateTime": {
          "earliest": "2020-07-04T00:00:00+02:00",
          "latest": "2021-07-04T00:00:00+02:00"
        }
      }
    }
  },
  {
    "name": "end_date",
    "display_name": "end_date",
    "base_type": "type/Date",
    "field_ref": [
      "field",
      "end_date",
      {
        "base-type": "type/Date"
      }
    ],
    "semantic_type": null,
    "fingerprint": {
      "global": {
        "distinct-count": 2,
        "nil%": 0
      },
      "type": {
        "type/DateTime": {
          "earliest": "2020-07-10T00:00:00+02:00",
          "latest": "2021-07-10T00:00:00+02:00"
        }
      }
    }
  },
  {
    "name": "rank",
    "display_name": "rank",
    "base_type": "type/Integer",
    "field_ref": [
      "field",
      "rank",
      {
        "base-type": "type/Integer"
      }
    ],
    "semantic_type": null,
    "fingerprint": {
      "global": {
        "distinct-count": 2,
        "nil%": 0
      },
      "type": {
        "type/Number": {
          "min": 1,
          "q1": 1.614142842854285,
          "q3": 2,
          "max": 2,
          "sd": 0.28232985128663995,
          "avg": 1.9166666666666667
        }
      }
    }
  },
  {
    "name": "campaign_product",
    "display_name": "campaign_product",
    "base_type": "type/Text",
    "field_ref": [
      "field",
      "campaign_product",
      {
        "base-type": "type/Text"
      }
    ],
    "semantic_type": null,
    "fingerprint": {
      "global": {
        "distinct-count": 13,
        "nil%": 0
      },
      "type": {
        "type/Text": {
          "percent-json": 0,
          "percent-url": 0,
          "percent-email": 0,
          "percent-state": 0,
          "average-length": 12.291666666666666
        }
      }
    }
  },
  {
    "name": "clicks",
    "display_name": "clicks",
    "base_type": "type/Decimal",
    "field_ref": [
      "field",
      "clicks",
      {
        "base-type": "type/Decimal"
      }
    ],
    "semantic_type": null,
    "fingerprint": {
      "global": {
        "distinct-count": 24,
        "nil%": 0
      },
      "type": {
        "type/Number": {
          "min": 176,
          "q1": 490,
          "q3": 12528,
          "max": 71302,
          "sd": 16784.45552600226,
          "avg": 9506
        }
      }
    }
  },
  {
    "name": "cpc",
    "display_name": "cpc",
    "base_type": "type/Float",
    "field_ref": [
      "field",
      "cpc",
      {
        "base-type": "type/Float"
      }
    ],
    "semantic_type": null,
    "fingerprint": {
      "global": {
        "distinct-count": 24,
        "nil%": 0
      },
      "type": {
        "type/Number": {
          "min": 0.11838420465268676,
          "q1": 0.4014273303614404,
          "q3": 0.7476920630593793,
          "max": 1.839754098360656,
          "sd": 0.3649814116611265,
          "avg": 0.5999229976542102
        }
      }
    }
  },
  {
    "name": "client_cost",
    "display_name": "client_cost",
    "base_type": "type/Float",
    "field_ref": [
      "field",
      "client_cost",
      {
        "base-type": "type/Float"
      }
    ],
    "semantic_type": "type/Cost",
    "fingerprint": {
      "global": {
        "distinct-count": 24,
        "nil%": 0
      },
      "type": {
        "type/Number": {
          "min": 150.67,
          "q1": 371.65999999999997,
          "q3": 1674.2414814999997,
          "max": 26839.06391300001,
          "sd": 7130.776733207882,
          "avg": 3756.494603333334
        }
      }
    }
  },
  {
    "name": "conversions",
    "display_name": "conversions",
    "base_type": "type/Float",
    "field_ref": [
      "field",
      "conversions",
      {
        "base-type": "type/Float"
      }
    ],
    "semantic_type": null,
    "fingerprint": {
      "global": {
        "distinct-count": 24,
        "nil%": 0
      },
      "type": {
        "type/Number": {
          "min": 0,
          "q1": 6.82,
          "q3": 324.72,
          "max": 2428.6699999999996,
          "sd": 627.8836861679057,
          "avg": 333.57083333333327
        }
      }
    }
  },
  {
    "name": "conversion_rate",
    "display_name": "conversion_rate",
    "base_type": "type/Float",
    "field_ref": [
      "field",
      "conversion_rate",
      {
        "base-type": "type/Float"
      }
    ],
    "semantic_type": "type/Share",
    "fingerprint": {
      "global": {
        "distinct-count": 24,
        "nil%": 0
      },
      "type": {
        "type/Number": {
          "min": 0,
          "q1": 0.009874879525622686,
          "q3": 0.03422960101749244,
          "max": 0.09385174203113417,
          "sd": 0.02167518640302775,
          "avg": 0.025823456615265566
        }
      }
    }
  },
  {
    "name": "cpa",
    "display_name": "cpa",
    "base_type": "type/Float",
    "field_ref": [
      "field",
      "cpa",
      {
        "base-type": "type/Float"
      }
    ],
    "semantic_type": null,
    "fingerprint": {
      "global": {
        "distinct-count": 24,
        "nil%": 0.04166666666666667
      },
      "type": {
        "type/Number": {
          "min": 1.685812678704011,
          "q1": 16.94993473459159,
          "q3": 47.09510550996483,
          "max": 121.55666666666666,
          "sd": 31.151946659164196,
          "avg": 36.03222309347717
        }
      }
    }
  },
  {
    "name": "avg_revenue",
    "display_name": "avg_revenue",
    "base_type": "type/Float",
    "field_ref": [
      "field",
      "avg_revenue",
      {
        "base-type": "type/Float"
      }
    ],
    "semantic_type": null,
    "fingerprint": {
      "global": {
        "distinct-count": 24,
        "nil%": 0.04166666666666667
      },
      "type": {
        "type/Number": {
          "min": 74.10666666666667,
          "q1": 94.05944082726924,
          "q3": 114.69303093574898,
          "max": 170.2505353319058,
          "sd": 22.137459322489505,
          "avg": 109.37225484688331
        }
      }
    }
  },
  {
    "name": "revenue",
    "display_name": "revenue",
    "base_type": "type/Float",
    "field_ref": [
      "field",
      "revenue",
      {
        "base-type": "type/Float"
      }
    ],
    "semantic_type": null,
    "fingerprint": {
      "global": {
        "distinct-count": 24,
        "nil%": 0.04166666666666667
      },
      "type": {
        "type/Number": {
          "min": 222.32,
          "q1": 1192.07,
          "q3": 46491.350000000006,
          "max": 261699.09000000005,
          "sd": 69661.50477287096,
          "avg": 37980.4495652174
        }
      }
    }
  },
  {
    "name": "roas",
    "display_name": "roas",
    "base_type": "type/Float",
    "field_ref": [
      "field",
      "roas",
      {
        "base-type": "type/Float"
      }
    ],
    "semantic_type": null,
    "fingerprint": {
      "global": {
        "distinct-count": 24,
        "nil%": 0.04166666666666667
      },
      "type": {
        "type/Number": {
          "min": 0.6096470781802725,
          "q1": 2.827956066145653,
          "q3": 6.028465240608069,
          "max": 65.72925119709137,
          "sd": 16.971739676038858,
          "avg": 9.38380176404016
        }
      }
    }
  }
]

**collection_position: Empty**

Thank you for your assistance!

**Comment by lukeman** [comment]
I'm seeing this issue after upgrading from v0.40.0 -> v0.40.1, but sounds like that is a coincidence. I am fairly certain I did not see this issue on v0.39.4, though.

I have two similar questions originally from 1.5 years ago that show mostly the same metrics for two different slices of data (the second started as a copy of the first before minor column changes). These are both raw SQL and both have explicitly formatted columns with custom titles and appear on a dashboard together (where they still render just fine).

If I load Question A in the editor view, the columns duplicate when it renders. If I go in to Question B, the columns do not duplicate. The obvious difference I can see is Question A was last edited 5 months ago (I believe that was on 0.37.x). Question B was last edited on July 9th using v0.39.4 (the revision log indicates the viz settings updated in one of the four iterations I made that day). Prior to that update last Friday, Question B had last gotten an edit on the same day as Question A.

With just one data point I'm not sure that only questions serialized before a certain version are the ones displaying this bug, but it's the only correlation I have so far so I figured I'd pass that along.

**Comment by lukeman** [comment]
Affected questions are ending up with duplicate `visualization_settings['table.columns']` entries for each column. Here's a snip of the entries for a single column that was duped, one with a `fieldRef` of the form `[string, string, string]` and the other `[string, string, Object]`:

```
    "table.columns": [{
        "name": "project_id",
        "fieldRef": ["field-literal", "project_id", "type/Text"],
        "enabled": true
    }, ...
    {
        "name": "project_id",
        "fieldRef": ["field", "project_id", {
            "base-type": "type/Text"
        }],
        "enabled": true
    }, ...
    ]
```

It appears the dupes are showing up on questions previously saved with `fieldRef` values in the older style. The duplication can happen live in the editor and it can be saved as part of a revision to the card json (as was the case when I duplicated an affected question to poke around).

**Comment by ZohaSpark** [comment]
Hi, thanks a lot @lukeman for your comments, they were very helpful to understand why this happens. 

@flamber do you think something can be implemented to prevent that from happening, or do you think we should update each of our questions by ourselves?

Thanks a lot for your time and consideration.

**Comment by ZohaSpark** [comment]
Hello again @flamber ! Sorry if I am bothering you, I just wanted to know if you had an idea about how to fix this.? (instead of manually changing each questions' visualization settings. Even with the API, we could make mistakes.)

Thank you very much!

**Comment by flamber** [comment]
Like @lukeman pointed out, this is partially caused by the dimension changes in 0.39.0, but only since 0.40.0 did the double columns happen (likely caused by #16761).

A potential workaround for SQL questions would be to change the query to `select null` and run the query and then change the query back, which should then include all the columns only once (and using the new dimensions).
It's possible to write a query that removes orphaned dimensions, but it would depend on having an updated `result_metadata`, which cannot be guaranteed, so I cannot provide a workaround that will fix all existing questions without potentially breaking things. Sorry @ZohaSpark 

Reproduce:
1. On 0.38.4, create Native query > Sample Dataset > `select ID as "num", CATEGORY as "text" from PRODUCTS limit 1`
2. Move around with the columns, so we create `table.columns` - I just moved "num" to the end - save the question
![image](https://user-images.githubusercontent.com/1447303/126983358-86c5a533-ba3a-4c28-a40a-539c255995ae.png)
3. Upgrade to 0.40.1, and view the question - change `select ID as` to `select RATING as`. Before even running the query, it has now created duplicates of the columns
![image](https://user-images.githubusercontent.com/1447303/126983616-33034cf9-053e-4a38-85fa-4de317120e9d.png)
4. Run the query - it has now created an extra "num" column - besides the original and the duplicate
![image](https://user-images.githubusercontent.com/1447303/126983787-cc5a3479-9a19-4646-a873-6638b335b35f.png)

Alternative reproduce:
1. On 0.40.1, create Native query > Sample Dataset > `select ID as "num", CATEGORY as "text" from PRODUCTS limit 1`
2. Move around with the columns, so we create `table.columns` - I just moved "num" to the end
3. In the query, change `select ID as` to `select RATING as` and run the query - we now have an extra "num" column:
![image](https://user-images.githubusercontent.com/1447303/126984142-c14f9248-f212-4ec3-92fc-ab4aee60f42c.png)

Pinging @kulyk since he has a better understanding of visualization settings.
Giving P2 for now, but it might be a P1.

**Comment by camsaul** [comment]
@kulyk, any idea if this is something we can fix in the FE itself or do we need to do a fix for the result metadata in the backend too?

**Comment by daltojohnso** [comment]
@camsaul -- https://github.com/metabase/metabase/pull/17279 fixes the logic that was causing the duplications, but saved questions already affected by the bug won't be fixed; the FE has no way of knowing which duplicate column is the correct column.

**Comment by nemanjaglumac** [comment]
Dalton did such a great job of covering this with unit tests. Do you think it makes sense to add e2e test on top of that @flamber ?

**Comment by flamber** [comment]
@nemanjaglumac If you think it's impossible for this to happen again with the current test, then no - otherwise yes, always.
